File: Documentation/example_out.js
var buffer;

function init(gl)
{
   var myData = new DavidsFloatData([1., 2., 3., 4.]);

   buffer = gl.createBuffer(); // corresponding C function is glGenBuffers(GLsizei n, GLuint *buffers)
   gl.bindBuffer(GL_ARRAY_BUFFER, buffer);
   gl.bufferData(GL_ARRAY_BUFFER, DavidsSizeOf(myData), GL_STATIC_DRAW);
}

function draw(gl)
{
   gl.bindBuffer(GL_ARRAY_BUFFER, buffer);
}

function mouseMoved(e)
{
   // do something
}

File: Documentation/example_out.cpp
class GraphicsApp
{
   int bufferID;

   void init()
   {
      float buffer[] = {1., 2., 3., 4.};

      glGenBuffers(1, &bufferID);
      glBindBuffer(GL_ARRAY_BUFFER, bufferID);
      glBufferData(GL_ARRAY_BUFFER, sizeof(float) * 4, buffer, GL_STATIC_DRAW);
   }

   void draw()
   {
      glBindBuffer(GL_ARRAY_BUFFER, bufferID);
   }

   void mouseMoved(MouseEvent e)
   {
      // do something
   }
};
